src/itdelatrisu/opsu/crash/ErrorReport.java

```diff
@@ -0,0 +1,369 @@
+			StringWriter writer = new StringWriter();
+			cause.printStackTrace(new PrintWriter(writer));
+			return writer.toString();
+		}
+
+		return "No stack trace for the cause is available.";
+	}
+
+	/**
+	 * Compiles the report to a single string instance
+	 * 
+	 * @return The complete error report
+	 */
+	public String toString() {
+		StringBuilder builder = new StringBuilder();
+		// Header
+		builder.append("---- " + OpsuConstants.PROJECT_NAME + " Error Report ----");
+		builder.append("\n\n");
+		// Timestamp
+		builder.append("Timestamp: ");
+		builder.append((new SimpleDateFormat()).format(new Date()));
+		builder.append("\n");
+		// Error description
+		builder.append("Description: ");
+		builder.append(getDescription());
+		builder.append("\n\n");
+		// stack trace
+		builder.append(getCauseTrace());
+
+		// Additional information
+		builder.append("\n\nAdditional information:\n");
+		builder.append("----------------------------------------------------------------");
+		builder.append("\n\n");
+		appendSections(builder);
+		return builder.toString();
+	}
+
+	/**
+	 * Writes this error report to a separate file and returns it. If the report
+	 * was not written, return the log.
+	 */
+	public File writeToFile() {
+		File report = new File(Options.ERROR_REPORTS_DIR,
+				"error-report-" + (new SimpleDateFormat("yyyy-MM-dd_HH.mm.ss")).format(new Date()) + ".txt");
+		try {
+			report.getParentFile().mkdirs();
+			FileWriter writer = new FileWriter(report);
+			writer.write(toString());
+			writer.flush();
+			writer.close();
+		} catch (Exception e) {
+			Log.error("Cannot save error report");
+			report = Options.LOG_FILE;
+		}
+		return report;
+	}
+
+	/**
+	 * Posts the entire error report to
+	 * <a href="https://hastebin.com/about.md">hastebin</a> to avoid cluttering
+	 * the GitHub reporting system.
+	 * 
+	 * @return A formatted response for use in the error overview.
+	 * @see #getOverview()
+	 */
+	public String haste() {
+		try {
+			byte[] post = toString().getBytes(StandardCharsets.UTF_8);
+			HttpsURLConnection.setFollowRedirects(false);
+			// Open connection to hastebin's documents and POST the report
+			HttpsURLConnection conn = (HttpsURLConnection) new URL("https://hastebin.com/documents").openConnection();
+			conn.setRequestMethod("POST");
+			conn.setRequestProperty("User-Agent", "Mozilla/5.0");
+			conn.setRequestProperty("Content-Type", "text/plain");
+			conn.setRequestProperty("Content-Length", String.valueOf(post.length));
+
+			// Write the POST request
+			conn.setDoOutput(true);
+			OutputStream os = conn.getOutputStream();
+			os.write(post);
+			os.flush();
+			os.close();
+
+			if (conn.getResponseCode() != 200)
+				return "Remote server responded with \'" + conn.getResponseMessage() + "\'.";
+
+			BufferedReader reader = new BufferedReader(new InputStreamReader(conn.getInputStream()));
+			String reply = reader.readLine();
+
+			// Parse the JSON response manually
```

Comment:
The `org.json` package is available in this project. It would be preferable to use it instead of manually parsing it.